lib/chef/knife/bootstrap.rb

```diff
@@ -67,6 +67,11 @@ class Bootstrap < Knife
         :description => "The ssh gateway",
         :proc => Proc.new { |key| Chef::Config[:knife][:ssh_gateway] = key }
 
+      option :ssh_gateway_identity,
+        :long => "--ssh-gateway-identity SSH_GATEWAY_IDENTITY",
+        :description => "The SSH identity file used for gateway authentication",
+        :proc => Proc.new { |key| Chef::Config[:knife][:ssh_gateway_identity] = key }
```

Comment:
i don't think we actually need to do this proc dance any more provided that we use `config[:ssh_gateway_identity]` below

as long as cli options override the config file settings there's no reason for this.

at the same time i don't super care and this mistake is all over the knife config and we just haven't bothered to ever clean it all up.